### ISSUE TITLE
Extension recommendations should only be made if the user does not already have all recommended extensions

### ISSUE THREAD

**Issue by esetnik** [Microsoft/vscode#31345]
When I open vscode the editor shows a bar indicating there are extension recommendations, but I already have all recommended extensions installed. It should detect this scenario and prevent the recommendation banner.

- VSCode Version: Code 1.14.2 (cb82febafda0c8c199b9201ad274e25d9a76874e, 2017-07-19T23:22:50.650Z)
- OS Version: Darwin x64 16.7.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
xml|Dot|1.9.2
EditorConfig|Edi|0.9.4
material-icon-theme|PKi|2.0.11
vscode-docker|Pet|0.0.16
html-css-class-completion|Zig|1.8.0
html-snippets|abu|0.1.0
path-intellisense|chr|1.4.2
gitignore|cod|0.5.0
vscode-eslint|dba|1.2.11
python|don|0.6.9
vscode-babel-coloring|dza|0.0.4
vscode-npm-script|eg2|0.2.0
prettier-vscode|esb|0.21.0
php-debug|fel|1.11.1
php-intellisense|fel|1.4.3
docthis|joe|0.4.7
sublime-babel-vscode|jos|0.2.6
Go|luk|0.6.62
csharp|ms-|1.11.0
debugger-for-chrome|msj|3.1.6
php-docblocker|nei|1.2.0
Ruby|reb|0.12.1
sass-indented|rob|1.4.1
vim|vsc|0.9.0

(1 theme extensions excluded)

---

Steps to Reproduce:

1. Add `.vscode/extensions.json` with
```
{
	// See http://go.microsoft.com/fwlink/?LinkId=827846
	// for the documentation about the extensions.json format
	"recommendations": [
    // Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
    "dzannotti.vscode-babel-coloring",
    "msjsdiag.debugger-for-chrome",
    "EditorConfig.editorconfig",
    "dbaeumer.vscode-eslint",
    "esbenp.prettier-vscode"
	]
}
```
2. Launch vscode

Reproduces without extensions: Yes

**Comment by Dragon-Bytes** [comment]
agreed, and setting this off doesn't even work

```
  // Ignore extension recommendations
  "extensions.ignoreRecommendations": true,
```

Any reason why?

**Comment by esetnik** [comment]
Ignoring recommendations does work for me, so that may be a separate issue. However, for our use case, we version the extension recommendations and would only want to see this banner when new extensions the user doesn’t have are added to the list.

**Comment by Yohanna** [comment]
Hi @ramya-rao-a, this is still reproducible in `v1.19.0`, any chance you can take another look?

I still see this banner even though I already have the recommended extensions installed:
![image](https://user-images.githubusercontent.com/6926228/34212568-afb5e422-e56a-11e7-8da9-16653d2d4e18.png)

**Comment by ramya-rao-a** [comment]
Do you still see the prompt if you lowercase all the recommendations in the extensions.json file?

**Comment by Yohanna** [comment]
No, I don't! If they're all lowercase, everything works as expected.

**Comment by ramya-rao-a** [comment]
The issue of the prompt appearing when the recommendations in the `extensions.json` file are not all lowercase is fixed in the latest Insiders (1.20)
